src/hooks/core/useCoreChatSystemHook.ts

```diff
@@ -206,22 +218,34 @@ const useCoreChatSystemHook = () => {
           channelType[websocketData?.channel_type]
         );
 
-        queue.addPriorityJob({
-          operationLabel: DatabaseOperationLabel.CoreChatSystem_GeneralSystemMessage,
-          id: `${websocketData?.channel?.id}-${websocketData?.message?.id}`,
-          priority: QueueJobPriority.HIGH,
-          task: async () => {
-            if (
-              websocketData?.message?.message_type === 'notification-deleted' ||
-              websocketData?.message?.message_type === 'deleted'
-            ) {
-              await Promise.all([channelList.save(localDb)]);
-            } else {
-              const chat = ChatSchema.fromWebsocketObject(newWebsocketData);
-              await Promise.all([chat.save(localDb), channelList.save(localDb)]);
-            }
+        if (Platform.OS === 'android') {
+          if (
+            websocketData?.message?.message_type === 'notification-deleted' ||
+            websocketData?.message?.message_type === 'deleted'
+          ) {
+            await Promise.all([channelList.save(localDb)]);
+          } else {
+            const chat = ChatSchema.fromWebsocketObject(newWebsocketData);
+            await Promise.all([chat.save(localDb), channelList.save(localDb)]);
           }
-        });
+        } else {
+          queue.addPriorityJob({
+            operationLabel: DatabaseOperationLabel.CoreChatSystem_GeneralSystemMessage,
+            id: `${websocketData?.channel?.id}-${websocketData?.message?.id}`,
+            priority: QueueJobPriority.HIGH,
+            task: async () => {
+              if (
+                websocketData?.message?.message_type === 'notification-deleted' ||
+                websocketData?.message?.message_type === 'deleted'
+              ) {
+                await Promise.all([channelList.save(localDb)]);
+              } else {
+                const chat = ChatSchema.fromWebsocketObject(newWebsocketData);
+                await Promise.all([chat.save(localDb), channelList.save(localDb)]);
+              }
+            }
+          });
+        }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Similar to the previous hunk, this one also introduces platform-specific logic. The same considerations apply here: ensure that direct operations on Android do not block the UI thread.
```diff
+ if (Platform.OS === 'android') {
+   if (
+     websocketData?.message?.message_type === 'notification-deleted' ||
+     websocketData?.message?.message_type === 'deleted'
+   ) {
+     await Promise.all([channelList.save(localDb)]);
+   } else {
+     const chat = ChatSchema.fromWebsocketObject(newWebsocketData);
+     await Promise.all([chat.save(localDb), channelList.save(localDb)]);
+   }
+ } else {
+   queue.addPriorityJob({
+     operationLabel: DatabaseOperationLabel.CoreChatSystem_GeneralSystemMessage,
+     id: `${websocketData?.channel?.id}-${websocketData?.message?.id}`,
+     priority: QueueJobPriority.HIGH,
+     task: async () => {
+       if (
+         websocketData?.message?.message_type === 'notification-deleted' ||
+         websocketData?.message?.message_type === 'deleted'
+       ) {
+         await Promise.all([channelList.save(localDb)]);
+       } else {
+         const chat = ChatSchema.fromWebsocketObject(newWebsocketData);
+         await Promise.all([chat.save(localDb), channelList.save(localDb)]);
+       }
+     }
+   });
+ }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->